src/nwbinspector/checks/ecephys.py

```diff
@@ -79,3 +79,14 @@ def check_spike_times_not_in_unobserved_interval(units_table: Units, nunits: int
                     "observed intervals."
                 )
             )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=Units)
+def check_ascending_spike_times(units_table: Units):
+    """Check that the values in the timestamps array are strictly increasing."""
+    if "spike_times" not in units_table:
+        return
+    for spike_times_per_unit in units_table["spike_times"]:
+        differences_per_unit = np.diff(spike_times_per_unit)
+    if np.all(differences_per_unit >= 0):
+        return InspectorMessage(message=("This Units table contains spike times that are not ascending."))
```

Comment:
```suggestion
        if not np.all(spike_times_per_unit[:-1] <= spike_times_per_unit[1:]):
            return InspectorMessage(message=("This Units table contains spike times that are not ascending."))
```

A couple of suggestions here:

(i) from Ben's pointer, using the index shifting strategy instead (compare the array to a version of itself shifted by one index to the right)

(ii) indenting the `if` statement per unit inside the `for` loop

(iii) reversing the logic to say 'if not all entries are in sorted order' in line with (i); this should inherit an early return functionality where the first time it encounters a `False` in the array, it should exit